### PR TITLE
README: clarify defer/demand; typos

### PR DESCRIPTION
**Pull request by mralusw**
There was a typo. I've also tried to make `demand` more clear, as it seems some people shrink from using it (I was also confused by the docs).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -180,7 +180,7 @@ Multiple `config` blocks are also supported.
 ### Deferring plugin configuration
 
 With the introduction of the module system, some configurations have to be preformed after loading the module.
-The `defer` keyword is a shorthand to register a `ModuleLoaded` hook for given `module`.
+The `defer` keyword is a shorthand to register a `ModuleLoaded` hook for given `module`. You need to **`require` the module explicitly** elsewhere.
```
Comment (andreyorst) [inline README.md:183]:
```suggestion
The `defer` keyword is a shorthand to register a `ModuleLoaded` hook for given `module`.
You'll need to **`require` the module explicitly** elsewhere.
```

File: README.md [hunk 2]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
+    # config1 evaluated before demanding the module
     require-module fzf
-    # evaluated after demanding the module
+    # config2 evaluated after demanding the module
 }
 ```
 
 **Note**: the `ModuleLoaded` hook is defined as early as possible - before sourcing any of plugin files.
-The place where `require-module` call will be placed depends on the order of config blocks in the `plug` command.
+The place where `require-module` call will be placed depends on the order of config blocks in the `plug` command. As soon as the module is required, the `ModuleLoaded` hook will execute.
```
Comment (andreyorst) [inline README.md:225]:
```suggestion
The place where `require-module` call will be placed depends on the order of config blocks in the `plug` command.
As soon as the module is required, the `ModuleLoaded` hook will execute.
```

File: README.md [hunk 3]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
 Works the same as `defer` except requires the module immediately:
 
 ```kak
-plug "andreyorst/fzf.kak" config {
-    # evaluated before demanding the module
+plug "andreyorst/fzf.kak" config %{
+    # config1 evaluated before demanding the module
```
Comment (andreyorst) [inline README.md:204]:
```suggestion
    # config1 (evaluated before demanding the module)
```

File: README.md [hunk 4]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
 } demand fzf %{
-    set-option global fzf_project_use_tilda true
+    set-option global fzf_project_use_tilda true # demand block
```
Comment (andreyorst) [inline README.md:206]:
```suggestion
    # demand block (evaluated after module is required)
    set-option global fzf_project_use_tilda true 
```

File: README.md [hunk 5]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
 } config %{
-    # evaluated after demanding the module
+    # config2 evaluated after demanding the module
```
Comment (andreyorst) [inline README.md:208]:
```suggestion
    # config2 (evaluated after demanding the module)
```

File: README.md [hunk 6]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
 }
 ```
 
 The above snippet is a shorthand for this code:
 
 ``` kak
 plug "andreyorst/fzf.kak" defer fzf %{
-    set-option global fzf_project_use_tilda true
+    set-option global fzf_project_use_tilda true # demand block
```
Comment (andreyorst) [inline README.md:216]:
```suggestion
    # demand block (evaluated after module is required)
    set-option global fzf_project_use_tilda true
```

File: README.md [hunk 7]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
 } config %{
-    # evaluated before demanding the module
+    # config1 evaluated before demanding the module
```
Comment (andreyorst) [inline README.md:218]:
```suggestion
    # config1 (evaluated before demanding the module)
```

File: README.md [hunk 8]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
     require-module fzf
-    # evaluated after demanding the module
+    # config2 evaluated after demanding the module
```
Comment (andreyorst) [inline README.md:220]:
```suggestion
    # config2 (evaluated after demanding the module)
```

File: README.md [hunk 9]
```diff
@@ -200,29 +200,29 @@ plug "andreyorst/fzf.kak" config %{
...
 The above snippet is a shorthand for this code:
 
 ``` kak
 plug "andreyorst/fzf.kak" defer fzf %{
-    set-option global fzf_project_use_tilda true
+    set-option global fzf_project_use_tilda true # demand block
 } config %{
-    # evaluated before demanding the module
+    # config1 evaluated before demanding the module
     require-module fzf
```
Comment (andreyorst) [inline README.md:219]:
```suggestion
    require-module fzf # (the result of using 'demand' keyword)
```

**Review by andreyorst — CHANGES_REQUESTED** [review]
Thanks for the suggestions. I've made a few tweaks here and there.

Mainly, the style of writing in this document uses one sentence per line approach, as it makes it much easier to view git diffs later on. So I've suggested some edits to make your new sentences follow it.

Secondaly, I think yeah, it may be a bit unclear in current form, and your comments make it a bit clearer, but I think it should then be in the parentheses, as now `config1` and rest form the semantic structure, and the text is just a note

**Comment by mralusw** [comment]
Hi Andy, it's your baby, so please, change things how you like! Mine were just suggestions.

I've used a PR because that's the easiest contribution workflow, but I'm not looking to increase my commit count or anything :))

**Comment by andreyorst** [comment]
Fair enough :)

**Pull request merged**